srml/staking/src/lib.rs

```diff
@@ -1380,3 +1380,25 @@ impl<T: Trait> SelectInitialValidators<T::AccountId> for Module<T> {
 		<Module<T>>::select_validators().1
 	}
 }
+
+/// StakingSlasher
+pub struct StakingSlasher<T>(rstd::marker::PhantomData<T>);
+
+impl<T: Trait, Reporter>
```

Comment:
Why would the staking slasher be able to reward any kind of reporter? That seems overly polymorphic. Try to think a little further ahead. It's probably not possible to actually write rewarding code for reporter types that are not `T::AccountId`. At the very least it's non-obvious and we shouldn't set traps for future maintainers.

Reply:
Also, what if there are multiple reporters?

Reply:
it was of pure laziness because it is not used yet, shall probably be something like `IntoIterator<Item=T::AccountId>`

will fix